bundle/src/main/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImpl.java

```diff
@@ -0,0 +1,281 @@
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.jackrabbit.vault.fs.api.PathFilterSet;
+import org.apache.jackrabbit.vault.fs.filter.DefaultPathFilter;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+
+/**
+ * ACS AEM Commons - Asset Packager Servlet
+ * Servlet end-point used to create packages of pages and the referenced assets.
+ */
+@SuppressWarnings("serial")
+@SlingServlet(
+        methods = {"POST"},
+        resourceTypes = {"acs-commons/components/utilities/packager/asset-packager"},
+        selectors = {"package"},
+        extensions = {"json"}
+)
+public class AssetPackagerServletImpl extends AbstractPackagerServlet {
+
+    /* Property names */
+    private static final String PN_PAGE_PATH = "pagePath";
+    private static final String PN_EXCLUDE_PAGES = "excludePages";
+    private static final String PN_ASSET_PREFIX = "assetPrefix";
+    private static final String PN_PAGE_EXCLUSIONS = "pageExclusions";
+    private static final String PN_ASSET_EXCLUSIONS = "assetExclusions";
+
+    /* Default Properties */
+    private static final String DEFAULT_PACKAGE_NAME = "assets";
+    private static final String DEFAULT_PACKAGE_GROUP_NAME = "Assets";
+    private static final String DEFAULT_PACKAGE_DESCRIPTION = "Asset Package initially defined by a ACS AEM Commons - "
+            + "Asset Packager configuration.";
+    private static final String THUMBNAIL_RESOURCE_PATH =
+            "/apps/acs-commons/components/utilities/packager/asset-packager/definition/package-thumbnail.png";
+
+    @Reference
+    private PackageHelper packageHelper;
+
+    private String customPrefix;
+    private List<Pattern> pageExclusionPatterns;
+    private List<Pattern> assetExclusionPatterns;
+    private final List<String> excludedPages = new ArrayList<>();
+
+    @Override
+    public final void doPost(final SlingHttpServletRequest request,
+                             final SlingHttpServletResponse response) throws IOException {
+
+        final ResourceResolver resourceResolver = request.getResourceResolver();
+        final boolean preview = Boolean.parseBoolean(request.getParameter("preview"));
+
+        log.debug("Preview mode: {}", preview);
+
+        final ValueMap properties = this.getProperties(request);
+
+        this.customPrefix = properties.get(PN_ASSET_PREFIX, String.class);
+        this.pageExclusionPatterns = generatePatterns(properties.get(PN_PAGE_EXCLUSIONS, new String[]{}));
+        this.assetExclusionPatterns = generatePatterns(properties.get(PN_ASSET_EXCLUSIONS, new String[]{}));
+
+        try {
+            final String pagePath = properties.get(PN_PAGE_PATH, String.class);
+            final List<PathFilterSet> paths;
```

Comment:
Can we rename this to `pathFilters` to be more clear?

Reply:
or even `packagePathFilters`

Reply:
Renamed!